lib/common/client/concerns/monitoring.rb

```diff
@@ -14,14 +14,28 @@ def with_monitoring(trace_location = 1)
       increment_total(caller)
     end
 
+    private
+
     def increment_total(caller)
-      StatsD.increment("#{self.class::STATSD_KEY_PREFIX}.#{caller}.total")
+      increment("#{self.class::STATSD_KEY_PREFIX}.#{caller}.total")
     end
 
     def increment_failure(caller, error)
       tags = ["error:#{error.class}"]
       tags << "status:#{error.status}" if error.try(:status)
-      StatsD.increment("#{self.class::STATSD_KEY_PREFIX}.#{caller}.fail", tags: tags)
+
+      increment("#{self.class::STATSD_KEY_PREFIX}.#{caller}.fail", tags: tags)
+    end
+
+    def increment(tag)
+      # TODO this would poll redis for each call of with_monitoring.
```

Comment:
Each time we increment a metric, we will add that metric name to our running redis set correct?